### PR TITLE
lenovo-legion: 0.0.12 -> 0.0.18 with patches in master

### PR DESCRIPTION
**Pull request by CHN-beta**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by CHN-beta** [comment]
> Is there a reason we don't use 0.0.18 tag and then apply the patches instead of pointing at the commit?

I first tried version string `0.0.18-e9c1d8` and it failed to compile (the version string is written into `setup.cfg` and then makes it fail to compile). After that, I leave `0.0.18`.

Commits between current master (`e9c1d81`) and 0.0.18 are simply some document change and the patch for kernel 6.11. Manually applying the patch needs several additional lines of nix code, and just gives nearly identical code comparing to current master. I think it might be not worth to make the nix code complex.

**Comment by i-am-logger** [comment]
> Is there a reason we don't use 0.0.18 tag and then apply the patches instead of pointing at the commit? I can't remember what is preferable but it seems wrong to say that it's 0.0.18 when it's not

yes, i agree, this is what I initially did...though I probably did something wrong with my patch (locally)

**Review by ulrikstrid — APPROVED** [review]
I don't have access to my Legion laptop currently so can't test this for real, but the changes seems fine. I'm just waiting for my review bot to run the nixpkgs-review before we merge this.

**Comment by CHN-beta** [comment]
> 2 packages failed to build:
> 
>     * lenovo-legion
> 
>     * lenovo-legion.dist

But it built just well on my machine. Build log:

<details>

```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Running phase: qtPreHook
@nix { "action": "setPhase", "phase": "qtPreHook" }
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/gxprayzqiqikai10iw9zhlggja4pb5da-source
source root is source/python/legion_linux
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/python/legion_linux/setup.py
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing setuptoolsBuildPhase
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/legion_linux
copying legion_linux/__init__.py -> build/lib/legion_linux
copying legion_linux/legion.py -> build/lib/legion_linux
copying legion_linux/legion_cli.py -> build/lib/legion_linux
copying legion_linux/legion_gui.py -> build/lib/legion_linux
copying legion_linux/legion_logo.png -> build/lib/legion_linux
copying legion_linux/legion_logo_light.png -> build/lib/legion_linux
copying legion_linux/legion_logo_dark.png -> build/lib/legion_linux
/nix/store/hsidk8mv1c7pl3bn6b1g20a3nws36v7q-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/_distutils/cmd.py:66: SetuptoolsDeprecationWarning: setup.py install is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` directly.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
        ********************************************************************************

!!
  self.initialize_options()
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/legion_linux
copying build/lib/legion_linux/__init__.py -> build/bdist.linux-x86_64/wheel/legion_linux
copying build/lib/legion_linux/legion.py -> build/bdist.linux-x86_64/wheel/legion_linux
copying build/lib/legion_linux/legion_cli.py -> build/bdist.linux-x86_64/wheel/legion_linux
copying build/lib/legion_linux/legion_gui.py -> build/bdist.linux-x86_64/wheel/legion_linux
copying build/lib/legion_linux/legion_logo.png -> build/bdist.linux-x86_64/wheel/legion_linux
copying build/lib/legion_linux/legion_logo_light.png -> build/bdist.linux-x86_64/wheel/legion_linux
copying build/lib/legion_linux/legion_logo_dark.png -> build/bdist.linux-x86_64/wheel/legion_linux
running install_data
creating build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data
creating build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data
creating build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share
creating build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/pixmaps
copying legion_linux/legion_logo.png -> build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/pixmaps/
copying legion_linux/legion_logo_light.png -> build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/pixmaps/
copying legion_linux/legion_logo_dark.png -> build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/pixmaps/
creating build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/applications
copying legion_linux/legion_gui.desktop -> build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/applications/
creating build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/polkit-1
creating build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/polkit-1/actions
copying legion_linux/legion_cli.policy -> build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/polkit-1/actions/
copying legion_linux/legion_gui.policy -> build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/polkit-1/actions/
creating build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/legion_linux
copying legion_linux/extra/service/legiond.ini -> build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/legion_linux/
copying legion_linux/extra/service/profiles/balanced-ac.yaml -> build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/legion_linux/
copying legion_linux/extra/service/profiles/balanced-battery.yaml -> build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/legion_linux/
copying legion_linux/extra/service/profiles/balanced-performance-ac.yaml -> build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/legion_linux/
copying legion_linux/extra/service/profiles/balanced-performance-battery.yaml -> build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/legion_linux/
copying legion_linux/extra/service/profiles/performance-ac.yaml -> build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/legion_linux/
copying legion_linux/extra/service/profiles/performance-battery.yaml -> build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/legion_linux/
copying legion_linux/extra/service/profiles/quiet-ac.yaml -> build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/legion_linux/
copying legion_linux/extra/service/profiles/quiet-battery.yaml -> build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.data/data/share/legion_linux/
running install_egg_info
running egg_info
creating legion_linux.egg-info
writing legion_linux.egg-info/PKG-INFO
writing dependency_links to legion_linux.egg-info/dependency_links.txt
writing entry points to legion_linux.egg-info/entry_points.txt
writing requirements to legion_linux.egg-info/requires.txt
writing top-level names to legion_linux.egg-info/top_level.txt
writing manifest file 'legion_linux.egg-info/SOURCES.txt'
reading manifest file 'legion_linux.egg-info/SOURCES.txt'
writing manifest file 'legion_linux.egg-info/SOURCES.txt'
Copying legion_linux.egg-info to build/bdist.linux-x86_64/wheel/legion_linux-0.0.18-py3.12.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/legion_linux-0.0.18.dist-info/WHEEL
creating 'dist/legion_linux-0.0.18-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'legion_linux/__init__.py'
adding 'legion_linux/legion.py'
adding 'legion_linux/legion_cli.py'
adding 'legion_linux/legion_gui.py'
adding 'legion_linux/legion_logo.png'
adding 'legion_linux/legion_logo_dark.png'
adding 'legion_linux/legion_logo_light.png'
adding 'legion_linux-0.0.18.data/data/share/applications/legion_gui.desktop'
adding 'legion_linux-0.0.18.data/data/share/legion_linux/balanced-ac.yaml'
adding 'legion_linux-0.0.18.data/data/share/legion_linux/balanced-battery.yaml'
adding 'legion_linux-0.0.18.data/data/share/legion_linux/balanced-performance-ac.yaml'
adding 'legion_linux-0.0.18.data/data/share/legion_linux/balanced-performance-battery.yaml'
adding 'legion_linux-0.0.18.data/data/share/legion_linux/legiond.ini'
adding 'legion_linux-0.0.18.data/data/share/legion_linux/performance-ac.yaml'
adding 'legion_linux-0.0.18.data/data/share/legion_linux/performance-battery.yaml'
adding 'legion_linux-0.0.18.data/data/share/legion_linux/quiet-ac.yaml'
adding 'legion_linux-0.0.18.data/data/share/legion_linux/quiet-battery.yaml'
adding 'legion_linux-0.0.18.data/data/share/pixmaps/legion_logo.png'
adding 'legion_linux-0.0.18.data/data/share/pixmaps/legion_logo_dark.png'
adding 'legion_linux-0.0.18.data/data/share/pixmaps/legion_logo_light.png'
adding 'legion_linux-0.0.18.data/data/share/polkit-1/actions/legion_cli.policy'
adding 'legion_linux-0.0.18.data/data/share/polkit-1/actions/legion_gui.policy'
adding 'legion_linux-0.0.18.dist-info/METADATA'
adding 'legion_linux-0.0.18.dist-info/WHEEL'
adding 'legion_linux-0.0.18.dist-info/entry_points.txt'
adding 'legion_linux-0.0.18.dist-info/top_level.txt'
adding 'legion_linux-0.0.18.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Finished executing setuptoolsBuildPhase
Running phase: glibPreInstallPhase
@nix { "action": "setPhase", "phase": "glibPreInstallPhase" }
Running phase: glibPreInstallPhase
@nix { "action": "setPhase", "phase": "glibPreInstallPhase" }
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Executing pypaInstallPhase
Successfully installed legion_linux-0.0.18-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
@nix { "action": "setPhase", "phase": "pythonOutputDistPhase" }
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: qtOwnPathsHook
@nix { "action": "setPhase", "phase": "qtOwnPathsHook" }
Running phase: glibPreFixupPhase
@nix { "action": "setPhase", "phase": "glibPreFixupPhase" }
Running phase: glibPreFixupPhase
@nix { "action": "setPhase", "phase": "glibPreFixupPhase" }
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
shrinking RPATHs of ELF executables and libraries in /nix/store/4680670kj841jvdj65z69f5krrizyhzr-lenovo-legion-app-0.0.18
checking for references to /build/ in /nix/store/4680670kj841jvdj65z69f5krrizyhzr-lenovo-legion-app-0.0.18...
patching script interpreter paths in /nix/store/4680670kj841jvdj65z69f5krrizyhzr-lenovo-legion-app-0.0.18
stripping (with command strip and flags -S -p) in  /nix/store/4680670kj841jvdj65z69f5krrizyhzr-lenovo-legion-app-0.0.18/lib /nix/store/4680670kj841jvdj65z69f5krrizyhzr-lenovo-legion-app-0.0.18/bin
shrinking RPATHs of ELF executables and libraries in /nix/store/197vgsbp7q7w76w2sq8c8pq8ahaqwj1q-lenovo-legion-app-0.0.18-dist
checking for references to /build/ in /nix/store/197vgsbp7q7w76w2sq8c8pq8ahaqwj1q-lenovo-legion-app-0.0.18-dist...
patching script interpreter paths in /nix/store/197vgsbp7q7w76w2sq8c8pq8ahaqwj1q-lenovo-legion-app-0.0.18-dist
Rewriting #!/nix/store/l6z671x7a5sf5r5z6rfp9fn0vpl9k5d4-python3-3.12.5/bin/python3.12 to #!/nix/store/l6z671x7a5sf5r5z6rfp9fn0vpl9k5d4-python3-3.12.5
wrapping `/nix/store/4680670kj841jvdj65z69f5krrizyhzr-lenovo-legion-app-0.0.18/bin/legion_cli'...
Rewriting #!/nix/store/l6z671x7a5sf5r5z6rfp9fn0vpl9k5d4-python3-3.12.5/bin/python3.12 to #!/nix/store/l6z671x7a5sf5r5z6rfp9fn0vpl9k5d4-python3-3.12.5
wrapping `/nix/store/4680670kj841jvdj65z69f5krrizyhzr-lenovo-legion-app-0.0.18/bin/legion_gui'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
@nix { "action": "setPhase", "phase": "pythonCatchConflictsPhase" }
Running phase: pythonRemoveBinBytecodePhase
@nix { "action": "setPhase", "phase": "pythonRemoveBinBytecodePhase" }
Running phase: pythonImportsCheckPhase
@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
Executing pythonImportsCheckPhase
Running phase: postPatchMkspecs
@nix { "action": "setPhase", "phase": "postPatchMkspecs" }
```
</details>

**Comment by ulrikstrid** [comment]
https://stridbot.strid.ninja/nixpkgs-review/348417

This link might be broken in a couple of hours but shows the issue now. If you think it's a flakyness bug I could run it again.

**Comment by CHN-beta** [comment]
> If you think it's a flakyness bug I could run it again.

Build log shows the error is in python-pyqt6, not lenovo-legion. Maybe we should wait several hours or days for fixing on python-pyqt6 in current master, or we may run build test against current nixos-unstable?

**Comment by ulrikstrid** [comment]
> > If you think it's a flakyness bug I could run it again.
> 
> Build log shows the error is in python-pyqt6, not lenovo-legion. Maybe we should wait several hours or days for fixing on python-pyqt6 in current master, or we may run build test against current nixos-unstable?

Seems like pyqt6 is incompatible with qt 6.8 which is in master. Either we push for upstream to move to pyside or we wait for pyqt6 to release a compatible version, it looks like it's closed source so not easily fixed from our side.

I made a PR upgrading pyqt to 6.7.1 which is the latest but it's still broken: https://github.com/NixOS/nixpkgs/pull/348697

File: maintainers/maintainer-list.nix
Comment (eljamm) [inline maintainers/maintainer-list.nix:3807]: Update to the `maintainer-list` does not follow the [commit convention](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions):

> When adding yourself as maintainer in the same pull request, make a separate commit with the message maintainers: add <handle>. Add the commit before those making changes to the package or module. See [Nixpkgs Maintainers](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) for details.

While you're at it, you can also make separate commits for formatting and adding maintainers to the package as you should:

> Create a commit for each logical unit.

Which makes things clearer.

**Comment by Rexcrazy804** [comment]
> I'm not sure what is the best course of action here, this PR is likely fine but we're broken by a dependency that is broken on master.

image

from the upstream tags page, it seems like the switch from pyqt5 to pyqt6 happened at 0.0.16. Perhaps for the time being we could bump it form 0.0.12 to 0.0.15? [Assuming pyqt5 isn't broken]

**Comment by eljamm** [comment]
> from the upstream tags page, it seems like the switch from pyqt5 to pyqt6 happened at 0.0.16. Perhaps for the time being we could bump it form 0.0.12 to 0.0.15? [Assuming pyqt5 isn't broken].

I think it's just they forgot to remove `pyqt5` from the list of dependencies. The actual codebase switched to `pyqt6` happened much earlier in  https://github.com/johnfanv2/LenovoLegionLinux/commit/f5cf184d5faf80a26bb9d430df0d1a2cf2563c10

Rebasing this PR and the [pyqt 6.8 update PR](https://github.com/NixOS/nixpkgs/pull/348697) on top of master, I could successfully build `lenovo-legion`, so it seems to me that we just need to wait until that's merged.

Also, we should probably switch to building this using `qt6` and remove the dependency on `qt5` and `pyqt5`.

**Comment by Rexcrazy804** [comment]
> Rebasing this PR and the [pyqt 6.8 update PR](https://github.com/NixOS/nixpkgs/pull/348697) on top of master, I could successfully build `lenovo-legion`, so it seems to me that we just need to wait until that's merged.

That's great. but yeah I got the thing working with pyqt5 with the following 
https://gist.github.com/Rexcrazy804/192aaf6673f89b624c6997f7ba9e4e08

it builds and runs successfully*
image

\* I am currently not on my legion system so I can't see if the the program is able to read the data, but the gui stuff looks alright.

> I think it's just they forgot to remove pyqt5 from the list of dependencies. The actual codebase switch to pyqt6 happened much earlier in https://github.com/johnfanv2/LenovoLegionLinux/commit/f5cf184d5faf80a26bb9d430df0d1a2cf2563c10

yeah its mostly just changes to the deploy/ folders that contained instructions for build dependencies for the other distros and some minor changes to the gui.py file. We could make an inverse patch to get it working.

**Comment by eljamm** [comment]
Although patching will work, I think keeping `pyqt5` will only increase the maintenance effort, considering that upstream has switched to `pyqt6`, which means we might have to patch even more things in the future. Also, we'll eventually have to do the switch as `qt5` becomes deprecated.

File: maintainers/maintainer-list.nix
Comment (CHN-beta) [reply maintainers/maintainer-list.nix:3807]: Thank you for pointing out this. I will correct it tomorrow.

**Comment by CHN-beta** [comment]
Thank you to all the reviewers. I will solve all the mentioned issues tomorrow.

**Comment by ulrikstrid** [comment]
With pyqt6 update merged this builds okay. Are you okay with this @eljamm ?

**Comment by ulrikstrid** [comment]
Also @realsnick

**Comment by eljamm** [comment]
Yes, we just need to address the commit convention and remove the dependency on `pyqt5` and I think this would be good to merge.

File: maintainers/maintainer-list.nix
Comment (ulrikstrid) [reply maintainers/maintainer-list.nix:3807]: I'm pretty sure the convention should be `maintainers: add chn` but I could be wrong, please correct me @eljamm

Comment (eljamm) [reply maintainers/maintainer-list.nix:3807]: > I'm pretty sure the convention should be maintainers: add chn

Yes, that's correct.

Comment (CHN-beta) [reply maintainers/maintainer-list.nix:3807]: I will correct it now.

Comment (ulrikstrid) [reply maintainers/maintainer-list.nix:3807]: Thank you :heart:

**Review by ulrikstrid — APPROVED** [review]
LGTM!

File: maintainers/maintainer-list.nix
Comment (eljamm) [reply maintainers/maintainer-list.nix:3807]: Also, following the [versioning guide](https://github.com/NixOS/nixpkgs/blob/0a0c9e56cd7cfe1b18745fd5c6c741a3d1b6a0ca/pkgs/README.md#versioning), I think the package version and commit title should be changed from `0.0.18` to `"0.0.18-unstable-2024-09-22"` (and also remove `with patches in master` from the commit title)

File: pkgs/os-specific/linux/lenovo-legion/app.nix
Comment (eljamm) [inline pkgs/os-specific/linux/lenovo-legion/app.nix:11]: Following the [versioning convention](https://github.com/NixOS/nixpkgs/blob/0a0c9e56cd7cfe1b18745fd5c6c741a3d1b6a0ca/pkgs/README.md#versioning):

```suggestion
  version = "0.0.18-unstable-2024-09-22";
```

The commit title should also be changed to:
```
lenovo-legion: 0.0.12 -> 0.0.18-unstable-2024-09-22
```

Comment (eljamm) [reply pkgs/os-specific/linux/lenovo-legion/app.nix:11]: Nevermind, this apparently isn't working...

```
  File "/nix/store/l7idy2qiiv0v0b6khfjvz3l5k6mnm47l-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/_vendor/packaging/version.py", line 202, in __init__
    raise InvalidVersion(f"Invalid version: '{version}'")
packaging.version.InvalidVersion: Invalid version: '0.0.18-unstable-2024-09-22'
```

**Review by eljamm — APPROVED** [review]

File: pkgs/os-specific/linux/lenovo-legion/app.nix
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/lenovo-legion/app.nix:25]: That was wild and shouldn't have worked.

File: pkgs/os-specific/linux/lenovo-legion/app.nix
Comment (CHN-beta) [reply pkgs/os-specific/linux/lenovo-legion/app.nix:11]: > This should have been set but in postPatch we should only set 0.0.18

I have created a separate pr for this https://github.com/NixOS/nixpkgs/pull/349682

**Pull request merged**
